### PR TITLE
Fix SquareDal PDF

### PR DESCRIPTION
**Pull request by henryiii**
This fixes the use of Square Dalitz Eff (was not incrementing the parameter container when exiting early).

### CODE REVIEW

**Comment by marthaisabelhilton** [comment]
Hi @henryiii @danielsibemol I get the error non-positive normalisation when I try to use this pdf. Does it work for you? I have attached the minimal working example in which I set the constant term in the polynomial to one. 
[SquareDalitzEffPdf_test.py.txt](https://github.com/GooFit/GooFit/files/4077266/SquareDalitzEffPdf_test.py.txt)

File: tests/convert/CMakeLists.txt [hunk 1]
```diff
@@ -25,7 +25,7 @@ if(ROOT_FOUND)
     goofit_catch_test(ExpTest testhelpers.h)
     goofit_catch_test(ExpGaussTest testhelpers.h)
 #   goofit_catch_test(IncoherentSumTest testhelpers.h)
-    goofit_catch_test(InterHistTest testhelpers.h)
+#   goofit_catch_test(InterHistTest testhelpers.h)
```
Comment (henryiii) [inline tests/convert/CMakeLists.txt]: This causes GCC 9 to crash...

**Comment by henryiii** [comment]
This fixes the CI and a few other things, so let's go ahead and merge soon, then work on fixing the "non-positive normalisation" error. Agreed?

**Comment by danielsibemol** [comment]
I believe the non-positive normalization error is due to the choice of parameters, which were all set to 0 in the example. When using different parameters I don't get the error. I don't know if @marthaisabelhilton sees the same behaviour

**Pull request merged**